tests/unit/test_reportgenerator.py

```
@@ -357,3 +357,207 @@ def test_rank_anomalies_with_single_row(self, valid_config):
+        assert len(top_df) == len(ranked_df)
+        assert "exceeds DataFrame length" in caplog.text
+
+    def test_apply_tags_logs_flagged_count(
+        self, valid_config, sample_scored_df, caplog
+    ):
+        """Test apply_tags() logs the count of flagged customers."""
+        generator = ReportGenerator(valid_config)
+        ranked_df = generator.rank_anomalies(sample_scored_df)
+
+        with caplog.at_level(logging.INFO):
+            top_df = generator.apply_tags(ranked_df, top_n=20)
```

Ruff (0.14.5)

Local variable `top_df` is assigned to but never used

Remove assignment to unused variable `top_df`

(F841)